packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -707,6 +714,7 @@ export class ChainDefinitionBody extends DefinitionBody {
   public bind(scope: Construct, sfnPrincipal: iam.IPrincipal, sfnProps: StateMachineProps): DefinitionConfig {
     const graph = new StateGraph(this.chainable.startState, 'State Machine definition');
     graph.timeout = sfnProps.timeout;
+    graph.comment = sfnProps.comment;
```

Comment:
What if instead we do:

```ts
const graphJson = Stack.of(scope).toJsonString(graph.toGraphJson());
return {
  definitionString: { Comment: sfnProps.comment, ...graphJson },
};
```

Basically having the state machine augment the graph with properties that are only relevant to the state machine as a whole - and not to sub-graphs.

Reply:
finally updated,

```
    const graphJson = graph.toGraphJson();
    return {
      definitionString: Stack.of(scope).toJsonString({ ...graphJson, Comment: sfnProps.comment }),
    };
```

how's that? I included comment after since the graph always sets it (even if undefined) and {something: 'foo', ... {something:undefined}} will override with undefined